src/picologging/logrecord.cxx

```diff
@@ -94,9 +94,9 @@ int LogRecord_init(LogRecord *self, PyObject *initargs, PyObject *kwds)
         argsLen = PyObject_Length(args);
     }
 
-    if (argsLen == 1 && PySequence_Check(args)){
+    if (argsLen == 1 && PySequence_Check(args) && !PyUnicode_Check(args)){
         PyObject* firstValue = PySequence_GetItem(args, 0);
-        if (PyMapping_Check(firstValue)) {
+        if (PyMapping_Check(firstValue) && !PyUnicode_Check(firstValue)) {
```

Comment:
Considering this field is only accessed in `LogRecord_writeMessage` and used to call `PyUnicode_Format` which requires:

```cpp
/* Apply an argument tuple or dictionary to a format string and return
   the resulting Unicode string. */
```

We should just check dictionary and tuple.

Reply:
Made the change. For reference, here are the CPython discussions that led to the internal change from dict to collections.abc.mapping: 
https://bugs.python.org/issue21172
https://github.com/python/cpython/commit/1b7611405d80df5387cad6492b7b12436450cced

I think it's very unlikely that someone will send in something dict-like, but it could currently happen with stdlib. I could also add a test to ensure that if that does happen, it's not a seg fault and is more of a no-op.